src/Events/Custom_Tables/V1/WP_Query/Monitors/Query_Monitor.php

```diff
@@ -77,6 +86,37 @@ public function __construct( tad_DI52_Container $container = null ) {
 		$this->enabled = true;
 	}
 
+	/**
+	 * Will filter and retrieve the list of WP_Query_Modifier implementations. Any implementation filters
+	 * must be applied before the init hook is completed.
+	 *
+	 * @since TBD
+	 *
+	 * @return array<WP_Query_Modifier> List of WP_Query_Modifier implementations.
+	 */
+	public function get_implementations(): array {
+		// Keep running filter until init is finished.
+		if ( did_action( 'init' ) > 0 || empty( $this->implementations ) ) {
```

Comment:
```suggestion
		if ( doing_action( 'init' ) || ! did_action( 'init' ) ) {
```
Do no check for empty values: filtering code might have emptied the implementations for some legit reasons, just keep filtering until the `init` action has not been done.
Check for `doing_action ( 'init' )` as `did_action( 'init' )` will return `1` if the current action being dispatched is the `init` one, the value that will be read  by `did_action` is set before callbacks are applied.
Finally keep filtering if the `init` action has not been done yet.

Reply:
@lucatume I had similar to that originally but that logic failed because it was not run until after `init` - so it would never pass those conditions, hence the empty check e.g. "did it run yet?". 

I can check for null vs an array instead of empty? i.e. null means run it once at least, empty array skips.